File: android/app/src/main/java/es/manabe/yomiyasu/core/models/AccountModels.kt
package es.manabe.yomiyasu.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class RedeemRequest(
    val code: String,
    val username: String,
    val email: String,
    val password: String,
)

@Serializable
data class RedeemResponse(
    @SerialName("_id") val id: String,
    val username: String,
    val email: String,
    val admin: Boolean? = null,
)

@Serializable
data class UpdateUserRequest(
    val newUsername: String? = null,
    val oldPassword: String? = null,
    val newPassword: String? = null,
)

File: ios/Yomiyasu/Core/Models/AccountModels.swift
import Foundation

struct RedeemRequest: Encodable, Sendable {
    let code: String
    let username: String
    let email: String
    let password: String
}

struct RedeemResponse: Decodable, Sendable {
    let id: String
    let username: String
    let email: String
    let admin: Bool?

    private enum CodingKeys: String, CodingKey {
        case id = "_id"
        case username
        case email
        case admin
    }
}

struct UpdateUserRequest: Encodable, Sendable {
    var newUsername: String?
    var oldPassword: String?
    var newPassword: String?
}
